master/internal/api_experiment.go

```diff
@@ -0,0 +1,168 @@
+package internal
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/determined-ai/determined/master/internal/db"
+	"github.com/determined-ai/determined/master/pkg/actor"
+	"github.com/determined-ai/determined/master/pkg/model"
+	"github.com/determined-ai/determined/proto/pkg/apiv1"
+	"github.com/determined-ai/determined/proto/pkg/experimentv1"
+)
+
+var errExperimentNotFound = status.Error(codes.NotFound, "experiment not found")
+
+func canActivate(dbExp *model.Experiment) error {
+	if dbExp.State != model.PausedState {
+		return status.Error(codes.InvalidArgument, "experiment is not in a paused state")
+	}
+	return nil
+}
+
+func canPause(dbExp *model.Experiment) error {
+	if dbExp.State != model.ActiveState { // TODO does this make sense?
```

Comment:
blocking: You can only pause from an active state.

Reply:
let's chat about the best way to capture these possible transitions and state sets.